### PR TITLE
Add kotlin code for the chapter of sorting

### PR DESCRIPTION
**Pull request by curtishd**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.
屏幕截图 2024-03-16 081755
屏幕截图 2024-03-16 081805
屏幕截图 2024-03-16 081814
屏幕截图 2024-03-16 081822
屏幕截图 2024-03-16 081829
屏幕截图 2024-03-16 081836
屏幕截图 2024-03-16 081843
屏幕截图 2024-03-16 081901
屏幕截图 2024-03-16 081912

### CODE REVIEW

File: codes/kotlin/chapter_sorting/QuickSort.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * File: QuickSort.kt
+ * Created Time: 2024-1-25
+ * Author: curtishd (1023632660@qq.com)
+ */
+
+package chapter_sorting.QuickSort
+
+/* 元素交换 */
+fun swap(nums: IntArray, i: Int, j: Int) {
+    nums[i] = nums[j].also { nums[j] = nums[i] }
```
Comment (krahets) [inline codes/kotlin/chapter_sorting/QuickSort.kt:11]: Could you put all the quick sort functions into one file refer to other languages?

Comment (curtishd) [reply codes/kotlin/chapter_sorting/QuickSort.kt]: Sure！

File: codes/kotlin/chapter_sorting/quick_sort.kt [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    if (left >= right) return
+    // 哨兵划分
+    val pivot = partition(nums, left, right)
+    // 递归左子数组、右子数组
+    quickSort(nums, left, pivot - 1)
+    quickSort(nums, pivot + 1, right)
+}
+
+/* 选取三个候选元素的中位数 */
+fun medianThree(nums: IntArray, left: Int, mid: Int, right: Int): Int {
```
Comment (krahets) [inline codes/kotlin/chapter_sorting/quick_sort.kt:42]: This function has updates. Please synchronize the latest version

**Comment by krahets** [comment]
Hi @curtishd , I would recommend adding the auto-build workflow for Kotlin so that the code in each PR will be checked automatically.

You can check the workflow file of other languages at `https://github.com/krahets/hello-algo/tree/main/.github/workflows` and submit the workflow for Kotlin in another PR.

This is an example of an automatic code check:

image

**Comment by curtishd** [comment]
ok, I haven`t used it before. But I can have a try.

**Review by krahets — APPROVED** [review]
Thanks for the clean code!

**Pull request merged**